### PR TITLE
Iv oom

### PR DESCRIPTION
**Pull request by 35359595**
## Describe your changes

Clients are no longer blocked for message processing before runtime is available.

## Checklist before requesting a review
- [x] I have performed a self-review of my code.
- [x] If it is a core feature, I have added thorough tests.
- [x] I removed all Clippy and Formatting Warnings. 
- [x] I added required Copyrights.

### CODE REVIEW

**Review by Gauthamastro — DISMISSED** [review]

**Comment by nuel77** [comment]
orderbook worker is fetching runtime api. it is not waiting for pallet to be available now.
image

**Comment by 35359595** [comment]
> orderbook worker is fetching runtime api. it is not waiting for pallet to be available now. image

this is not a bug, it's intentional to allow client's messaging progress and release cache. we need to re-thing this snapshot loading on empty chain, possibly simply by demoting this error to warning.

File: clients/orderbook/src/worker.rs [hunk 1]
```diff
@@ -155,6 +157,8 @@ pub(crate) struct ObWorker<B: Block, BE, C, SO, N, R> {
 	last_processed_block_in_offchain_state: BlockNumber,
 	// Version of current state
 	state_version: Arc<RwLock<u16>>,
+	// is runtime upgraded and client can process
```
Comment (serhii-temchenko) [inline clients/orderbook/src/worker.rs]:
```suggestion
	/// is runtime upgraded and client can process
```
I'd suggest use doc syntax (e.g. `///`) here on pub scope as on other pub members...

File: clients/orderbook/src/worker.rs [hunk 2]
```diff
@@ -1575,13 +1572,23 @@ where
...
+						} else {
+							debug!(target: "orderbook", "📒 Waiting for orderbook pallet to become available...");
+							// dont process further
+							continue
+						}
 						if let Err(err) = self.handle_finality_notification(&finality).await {
 							error!(target: "orderbook", "📒 Error during finalized block import{}", err);
 						}
-					}else {
+					}	else {
```
Comment (serhii-temchenko) [inline clients/orderbook/src/worker.rs]:
```suggestion
					} else {
```

Code style. Extra space should be removed.

**Review by serhii-temchenko — CHANGES_REQUESTED** [review]

**Comment by Gauthamastro** [comment]
> > orderbook worker is fetching runtime api. it is not waiting for pallet to be available now. image
> 
> this is not a bug, it's intentional to allow client's messaging progress and release cache. we need to re-thing this snapshot loading on empty chain, possibly simply by demoting this error to warning.

It will result in an inconsistent state and there will be lots of changes in the worker logic, which we need to retest for corner cases. I will raise a PR with minimum changes

**Pull request closed without merging**